apps/web/src/pages/api/forms/registration.ts

```diff
@@ -3,26 +3,17 @@ import { logger } from '@nihr-ui/logger'
 import type { NextApiRequest, NextApiResponse } from 'next'
 import { ZodError } from 'zod'
 
+import { Wso2GroupsOperations } from '@/constants'
 import { REGISTRATION_CONFIRMATION_PAGE, REGISTRATION_PAGE } from '@/constants/routes'
 import { getUserWithRolesAndOrgs } from '@/lib/organisations'
 import { prismaClient } from '@/lib/prisma'
 import { isContactManagerAndSponsorContact, isSponsorContact } from '@/utils/auth'
 import type { RegistrationInputs } from '@/utils/schemas'
 import { registrationSchema } from '@/utils/schemas'
 
-/**
- * Next.js API route for handling user registration and creation in Identity Gateway (IDG).
- * This route expects a POST request with registration data and validates it.
- * If registration is successful, it creates a new user in IDG and updates the local user data.
- * If registration fails, it redirects back to the registration page with appropriate error messages.
- *
- * @param req - The HTTP request object containing registration data.
- * @param res - The HTTP response object for sending responses.
- */
-
 export async function assignRoleToUser(email: string, role: string) {
   try {
-    await authService.assignWSO2UserRole(email, role)
+    await authService.updateWSO2UserRole(email, role, Wso2GroupsOperations.Add)
```

Comment:
Mixing the Role vs Group naming here, so probably best to use the existing Role name in this layer.